src/ensembl/src/services/api-service.ts

```diff
@@ -67,12 +67,28 @@ class ApiService {
 
     try {
       const response = await fetch(url, fetchOptions);
+      if (!response.ok) {
+        throw await this.handleError(response, fetchOptions);
+      }
       return await this.handleResponse(response, fetchOptions);
     } catch (error) {
       throw error;
     }
   }
 
+  private async handleError(response: Response, options: FetchOptions) {
+    let errorJson;
+    try {
+      errorJson = await response.json();
+    } catch {
```

Comment:
As `catch` is empty how about removing it and moving the `return` statement into a `finally` clause?

Reply:
ok